SISRP-40774 - Final Exams Schedule (Instructors) - UX Dev

redconfetti opened the pull request:
https://jira.berkeley.edu/browse/SISRP-40774

lyttam commented on app/models/edo_oracle/queries.rb:
```diff
@@ -326,7 +326,7 @@ def self.get_section_final_exams(term_id, section_id)
         FROM
           SISEDO.CLC_FINAL_EXAM_INFOV00_VW exam
-        LEFT JOIN SISEDO.CLASSSECTIONALLV01_MVW sec ON (
+        RIGHT JOIN SISEDO.CLASSSECTIONALLV01_MVW sec ON (
```
@redconfetti why use a `RIGHT JOIN` here? Do you want to select class sections that don't have an exam?

redconfetti replied:
@lyttam - Yes. We need the 'exam_type' field from the section table, even if there isn't an exam. I've adjusted the code to account for blank exam fields.

lyttam replied:
@redconfetti ok. Just asking because that's a fundamental change to how the views are joined... before it was "select exams regardless of whether they have a section" and now it's "select sections regardless of whether they have an exam".

I would swap the views around and do a left join, just because I think that's more common and easier to read:
```
SELECT ...
FROM
  SISEDO.CLASSSECTIONALLV01_MVW sec
LEFT JOIN SISEDO.CLC_FINAL_EXAM_INFOV00_VW exam
...
```

lyttam commented on app/models/my_academics/exams.rb:
```diff
@@ -133,22 +156,33 @@ def parse_exam(exam)
+      if instructor_format
+        format = '%a, %b %-d'
+        format = '%b %-d, %Y' if !time_is_current_year?(date)
+      end
```
@redconfetti you could reduce the number of assignment operations being done here. One way to do it:

```
format = '%a %b %-d' unless instructor_format
format = '%a, %b %-d' if instructor_format && time_is_current_year?(date)
format = '%b %-d, %Y' if instructor_format && !time_is_current_year?(date)
```

redconfetti commented:
@lyttam - I've implemented the adjustments you suggested. I've also introduced a new directive for the display of Final Exam schedules on the Class Page.

redconfetti commented on src/assets/javascripts/angular/directives/academicsClassInfoFinalExamsDirective.js:
```diff
@@ -0,0 +1,43 @@
+var angular = require('angular');
+var _ = require('lodash');
+
+angular.module('calcentral.directives').directive('ccAcademicsClassInfoFinalExamsDirective', function() {
```
@lyttam - I didn't want to continue putting logic in `academicsController` just to service this one area of the class info page, so I made a directive. We should probably try to do this anywhere we can within the "My Academics" domain of pages/widgets (main page, semesters page, class page), so that the scope of dependencies is isolated and that we're closer to being ready to migrate to React components.

Would you agree?

lyttam commented:
@redconfetti Cool, I like the use of a directive. Looks good.
